src/Raven.Server/ServerWide/Commands/Subscriptions/PutSubscriptionCommand.cs

```diff
@@ -59,8 +61,16 @@ public override unsafe void Execute(TransactionOperationContext context, Table i
                     var existingSubscriptionState = JsonDeserializationClient.SubscriptionState(doc);
 
                     if (SubscriptionId != existingSubscriptionState.SubscriptionId)
+                    {
+                        if (string.IsNullOrEmpty(OldName))
+                        {
+                            SubscriptionName = $"{SubscriptionName}.{new Random().Next()}";
```

Comment:
Let's use something better here. This will generate a double.
Either use a Guid or an incrementing number